core/src/executor/select/mod.rs

```diff
@@ -124,8 +124,35 @@ pub async fn select_with_labels<'a>(
         order_by,
     } = match &query.body {
         SetExpr::Select(statement) => statement.as_ref(),
-        _ => {
-            return Err(SelectError::Unreachable.into());
+        SetExpr::Values(Values(values_list)) => {
+            let limit = Limit::new(query.limit.as_ref(), query.offset.as_ref())?;
+            let labels = values_list
+                .iter()
+                .enumerate()
+                .map(|(i, _)| format!("column{}", i + 1))
+                .collect::<Vec<_>>();
+            let first_len = values_list[0].len();
+            let rows = values_list
+                .iter()
+                .map(|exprs| {
+                    if exprs.len() != first_len {
+                        return Err(RowError::NumberOfValuesDifferent.into());
+                    }
+                    let values = exprs
+                        .iter()
+                        .map(|expr| evaluate_stateless(None, expr))
+                        .filter_map(Result::ok)
+                        .map(|evaluated| evaluated.try_into())
+                        .filter_map(Result::ok)
+                        .collect::<Vec<_>>();
```

Comment:
If error occurs in `evaluate_stateless` or `evaluated.try_into()`, then the current code will filter out and return `Ok` because of `filter_map`.
Rather than filter_map, it would be good to consider replacing `filter_map` to `and_then` like ones.
Then, by `collect::<Result<Vec<_>>>()` we can propagate `Result` to outside.